src/Model/Payment.php

```diff
@@ -559,9 +570,11 @@ protected function generateUniquePaymentIdentifier()
     {
         $generator = Injector::inst()->create(RandomGenerator::class);
         $id = null;
+        $length = static::config()->get('transaction_id_length');
```

Comment:
Same here. I think it's a good idea, but should not be part of this PR. Also, maybe a token configuration per gateway could be beneficial? Who knows what silly requirements other gateways have.